src/main/scala/nodes/learning/KernelGenerator.scala

```diff
@@ -0,0 +1,273 @@
+package nodes.learning
+
+import breeze.linalg._
+import breeze.math._
+import breeze.numerics._
+import org.apache.spark.rdd.RDD
+import org.apache.spark.SparkConf
+import org.apache.spark.SparkContext
+import org.apache.spark.SparkContext._
+import utils._
+
+
+// Trait to be implemented by Kernel functions
+trait KernelGenerator[T] {
```

Comment:
If we think of kernels as a feature transformation, then `KernelGenerator` can be a Transformer ? @sparks Its fine for Transformers to run jobs right ?